xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -864,6 +855,7 @@ glusterd_volinfo_delete(glusterd_volinfo_t *volinfo)
     int32_t ret = -1;
 
     GF_ASSERT(volinfo);
+    GF_ASSERT(GF_ATOMIC_GET(volinfo->refcnt) == 0);
```

Comment:
We should not introduce this check because glusterd_volinfo_delete is not call by only glusterd_volinfo_unref.
The other functions are also calling this function (glusterd_volinfo_delete) to delete a volume.

Reply:
If `glusterd_volinfo_t` is a ref-counted object, probably it's not correct to directly call `glusterd_volinfo_delete()`. How are those callers sure that there are no other active users of the volinfo ? However, even if they are sure they are the only user, `refcnt` should be `1`, so calling `glusterd_volinfo_unref()` should have the same effect and it's a more consistent approach IMO.

Reply:
> We should not introduce this check because glusterd_volinfo_delete is not call by only glusterd_volinfo_unref.

I'll check this. Anyway this is a bad practice because it breaks the reference counting as a concept at all.

Reply:
> > We should not introduce this check because glusterd_volinfo_delete is not call by only glusterd_volinfo_unref.
> 
> I'll check this. Anyway this is a bad practice because it breaks the reference counting as a concept at all.

Then we have to replace glusterd_volinfo_delete by glusterd_volinfo_unref.